### ISSUE TITLE
[Making the Grade] Union return type in Task 6

### ISSUE THREAD

**Issue by pranasziaukas** [exercism/python#2661]
https://github.com/exercism/python/blob/650a597311104f94f5c807a68c7859e16dc76327/exercises/concept/making-the-grade/loops.py#L51

IMO the union return type is not needed in a concept exercise. Besides, I don't see the value it provides at all in the current setting because `<student name>` is the only unique part that is matched.

My suggestion is to have a `str` return type, i.e.:
```
    :return: str - name of the first student with a perfect score or 'No perfect score.' if not found.
```

Happy to submit a PR.

**Comment by J08K** [comment]
Hey there, 

If we would look at it from a _broad_ view, yes, it wouldn't really make that much sense to teach a student to return a `Union` of two different types. **But**, I do have to say that when you look at the information given and the task at hand, it is easier to just return a `sublist` of a `list`. 

I think the exercise should just be about the concept of `loops` and that we shouldn't really try to overthrow that with much more information for the student to handle.

Anyways, that is just my opinion, I'd love to hear your comment on this @BethanyG and @pranasziaukas

**Comment by pranasziaukas** [comment]
> it is easier to just return a `sublist` of a `list`.

I'm not sure if this is what you had in mind @J08K but I agree that it is easy to return a sublist of students with perfect scores. I.e. `[[<student name A>, 100], [<student name B>, 100], ...]`, and a no-match would naturally be `[]`.
In fact, it is conceptually very similar to `above_threshold(student_scores, threshold=100)`.

That said, I think it's a good exercise to only ask for the first match.
Remember [`str.index`](https://docs.python.org/3/library/stdtypes.html?highlight=str#str.index) and [`str.find`](https://docs.python.org/3/library/stdtypes.html?highlight=str#str.find) methods. Both look for the (index of the) first match, and return either `ValueError` or out-of-bounds value `-1` upon failure.
I'm not suggesting to start throwing exceptions in this exercise, but an out-of-bounds `[]` would do the job with no unions needed. I think that would be more in line with standard python practices.

TL;DR I think
```
    :return: first `[<student name>, 100]` or `[]` if not found.
```
works as well.

Maybe I'm overthinking it all @BethanyG ¯\_(ツ)_/¯

**Comment by pranasziaukas** [comment]
I expect to pick this right after #2668 is merged (or dismissed).
